fe/fe-core/src/main/java/com/starrocks/connector/kudu/KuduPredicateConverter.java

```diff
@@ -0,0 +1,232 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.kudu;
+
+import com.google.common.collect.Lists;
+import com.starrocks.analysis.BinaryType;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CastOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.CompoundPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ConstantOperator;
+import com.starrocks.sql.optimizer.operator.scalar.InPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.IsNullPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.LikePredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperatorVisitor;
+import org.apache.kudu.ColumnSchema;
+import org.apache.kudu.Schema;
+import org.apache.kudu.client.KuduPredicate;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.apache.paimon.data.Timestamp.fromSQLTimestamp;
+
+public class KuduPredicateConverter extends ScalarOperatorVisitor<List<KuduPredicate>, Void> {
+    private static final Logger LOG = LogManager.getLogger(KuduPredicateConverter.class);
+    private final Schema schema;
+
+    public KuduPredicateConverter(Schema schema) {
+        this.schema = schema;
```

Comment:
I think there should be some predicates can't be converted, like `a+b > 5`, how shall we deal with this case?

Reply:
Limited by the kudu-client java library, it can only support a limited number of predicates, such as eq/lt/le/gt/ge/and/list, etc.

Reply:
> I think there should be some predicates can't be converted, like `a+b > 5`, how shall we deal with this case?

In this case, no predicate pushes down -,-

Reply:
I think you use the default predicate push down, so predicate like a+b>5 can be pushed down to SCAN_NODE, but we can't convert it to kudu predicate and i didn't find where we deal with it

Reply:
I'm not sure if I understand clearly, is it handled using KuduPredicateConverter#ExtractColumnName?